### PR TITLE
REVSDL-2597 OnServiceUnregistered notification GPB implementation

### PR DESCRIPTION
**Pull request by thinhts**
Please review
@KKarlash 
@DKlimenko 
@bonewell 
@VladSemenyuk 
@giangtqh

### CODE REVIEW

**Comment by KKarlash** [comment]
void VRModule::RemoveAppExtension(uint32_t app_id) {
  LOG4CXX_AUTO_TRACE(logger_);
  if (-1 != vector_of_registered_apps.find(app_id)) {
    OnUnregisterNotification();
    if (app_id == activated_connection_key())
      DeactivateService();
  }
}

**Comment by thinhts** [comment]
I updated. Please review
@KKarlash 
@bonewell

File: src/components/vr_cooperation/src/vr_module.cc [hunk 1]
```diff
@@ -363,5 +366,34 @@ void VRModule::ResetDefaultService() {
...
+  if (std::find(registered_apps_.begin(), registered_apps_.end(), app_id)
+      == registered_apps_.end()) {
+    registered_apps_.push_back(app_id);
+  }
+}
+
+void VRModule::UnregisterService(int32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::vector<int32_t>::iterator it = registered_apps_.begin();
+  for (it = registered_apps_.begin(); it != registered_apps_.end(); ++it) {
```
Comment (KKarlash) [inline src/components/vr_cooperation/src/vr_module.cc]: std::vector::iterator it = std::find(vector_of_registered_apps.begin(), vector_of_registered_apps.end(), app_id);
if (vector_of_registered_apps.end() != it) {
vector_of_registered_apps.erase(it);
}

**Comment by KKarlash** [comment]
Reviewed

**Pull request merged**